src/Security/OpenIdLoginAuthenticator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Security;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AbstractGuardAuthenticator;
+
+class OpenIdLoginAuthenticator extends AbstractGuardAuthenticator
+{
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    public function __construct(SessionInterface $session)
+    {
+        $this->session = $session;
+    }
+
+    public function supports(Request $request)
+    {
+        if (!$request->query->has('state') || !$request->query->has('id_token')) {
+            return false;
+        }
+
+        if ($request->query->get('state') !== $this->session->get('oauth2state')) {
+            $this->session->remove('oauth2state');
+            throw new \RuntimeException('Invalid state');
+        }
+
+        return true;
+    }
+
+    public function getCredentials(Request $request)
+    {
+        $code = $request->query->get('id_token');
+        [$jose, $payload, $signature] = array_map('base64_decode', explode('.', $code));
```

Comment:
Add a comment telling what's going on here, e.g. by adding a url to the jwt specification.